aiohttp/client.py

```diff
@@ -490,10 +491,13 @@ async def _request(
         if url.scheme not in self._connector.allowed_protocol_schema_set:
             raise NonHttpUrlClientError(url)
 
-        skip_headers = set(self._skip_auto_headers)
+        skip_headers: Optional[Union[frozenset[istr], set[istr]]] = None
         if skip_auto_headers is not None:
-            for i in skip_auto_headers:
-                skip_headers.add(istr(i))
+            skip_headers = {istr(i) for i in skip_auto_headers}
+            if self._skip_auto_headers is not None:
+                skip_headers.update(self._skip_auto_headers)
```

Comment:
Looks like we don't have coverage for passing `skip_auto_headers` and also setting it on the session